expression-language/src/main/java/io/zeebe/el/impl/FeelExpressionLanguage.java

```diff
@@ -47,22 +46,12 @@ public Expression parseExpression(final String expression) {
     ensureNotNull("expression", expression);
 
     final var expressionMatcher = EXPRESSION_PATTERN.matcher(expression);
-    final var valueMather = STATIC_VALUE_PATTERN.matcher(expression);
 
     if (expressionMatcher.matches()) {
       final var unpackedExpression = expressionMatcher.group(1);
       return parseFeelExpression(unpackedExpression);
-
-    } else if (valueMather.matches()) {
-      final var value = valueMather.group();
-      return new StaticExpression(value);
-
     } else {
-      final var failureMessage =
-          String.format(
-              "Expected FEEL expression (e.g. '=variableName') or static value (e.g. 'jobType') but found '%s'",
-              expression);
-      return new InvalidExpression(expression, failureMessage);
+      return new StaticExpression(expression);
```

Comment:
It might be good to check that the expression is not empty. This should be checked on validation but just to make sure...

Reply:
What is wrong about an empty exception?

Reply:
Do you see a case where an empty expression would be valid?

I think an empty expression indicates that the user forgot to fill something in. For example, how to deal with an empty job type.

Reply:
I don't see a case where an empty expression would be valid. I can imagine cases where maybe a value is optional and so an empty expression is benign.

My overall instinct is not to make any assumptions here. I think the validators are the perfect place to assess whether an empty expression can be valid. This place here could be called from anywhere. 

Speaking of validators, right now the `ZeebeExpressionValidator` will accept null and empty values:

```
  private void validateExpression(
      final String expression, final ValidationResultCollector resultCollector) {

    if (expression == null || expression.isEmpty()) {
      return;
    }
```
If you say, this should always be considered an invalid user input, we could change that. 

I also wonder whether long term we could have a connection between `ZeebeTaskDefinitionImpl#registerType` and `ZeebeRuntimeValidators`. I.e. maybe the runtime validators could be derived from the model constraints. 

Like I noticed in the retries issue that when I pass an empty expression, it falls back to the default value. So those two things seem somewhat related to me.

Reply:
Agree. The check for the empty expression was just about having a safeguard if the validation is removed for some reason. But it is not necessary. 

Regarding `ZeebeExpressionValidator`, I saw this too and changed it in my PR (https://github.com/zeebe-io/zeebe/pull/4079/files#diff-a00c815cabc9efbdbb0400dfc103be1aR64). It should verify that the expression is not null or empty - if there is no default value (as it is for the job retries).

I'm open to ideas about how to improve the validation. But better in a follow-up issue.